### PR TITLE
STYLE: Uncrustify Parallel-Explicit solvers

### PR DESCRIPTION
**Pull request by jacob-moore22**
# Description

There are a few changes associated with this PR:
- Uncrustify has been ran on all .cpp and .h files under Parallel-Explicit
- Uncrustify has been ran on all .cpp and .h files in the top-level Parallel-Solvers directory
- The .h and the .cpp files in the SGH solver have been separated into an include and a src file
- The CMakeLists.txt file has been updated to build with the modified file structure
- Some limited doxygen documentation has been added throughout. More to come
- Setup routines from the FEA_Module_SGH.cpp file have been pulled out and defined in a setup_sgh file
- Adds some crude functionality for automated integrated tests. More to come. 
- Adds an MSU option to build for machines administered on their network.
- Add uncrustify download and build via GitHub module
- Adds bash script to automate uncrustification process 
- Modified uncrustify config to add default uncrustify comment block to classes and functions

## Type of change

Please select all relevant options

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Formatting and/or style fixes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Test A : The code has been built from scratch and the SGH solver has been used to simulate the Sedov problem
- [x] Test B : The parallel implicit solver has been ran with TO active after the build. 

**Test Configuration**:
* OS version: Linux (Rocky)
* Hardware: Intel(R) Xeon(R) CPU E5-1630 v3 @ 3.70GHz
* Compiler: GCC 12.2.0

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] The code builds from scratch with my new changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by nathanielmorgan** [comment]
The improvements to the coding make it easier to read.

**Review by nathanielmorgan — APPROVED** [review]
Excellent improvements to the readability of the coding

File: src/Parallel-Solvers/Parallel-Explicit/example_simple.yaml
Comment (Adrian-Diaz) [inline src/Parallel-Solvers/Parallel-Explicit/example_simple.yaml:1]: Was the change to the default yaml file intended?

Comment (jacob-moore22) [reply src/Parallel-Solvers/Parallel-Explicit/example_simple.yaml:1]: I meant to change it back.  Ill push that fix now.

**Review by Adrian-Diaz — APPROVED** [review]

**Pull request merged**